### PR TITLE
add: support for https in Browser Sync

### PR DESCRIPTION
**Pull request by miklb**
address #6

### CODE REVIEW

**Comment by bamadesigner** [comment]
Hi @miklb! Thanks for the PR. Do you mind removing all of the indent changes so we can keep the PR limited to browserSync changes?

**Comment by miklb** [comment]
Sure. Didn't realize I messed up indents. I'll take a look at my linter settings and clean it up.

File: gulpfile.babel.js [hunk 1]
```diff
@@ -82,15 +82,26 @@ const server = browserSync.create();
 // Initialize the BrowserSync server conditionally:
 function serve(done) {
 	if (config.dev.browserSync.live) {
-		server.init({
-			proxy: config.dev.browserSync.proxyURL,
-			port: config.dev.browserSync.bypassPort,
-			liveReload: true
-		});
+		if (config.dev.browserSync.https) {
```
Comment (bamadesigner) [inline gulpfile.babel.js]: If we need to setup different arguments, let's start with the commonalities, e.g.:
`var serverConfig = {
    proxy: config.dev.browserSync.proxyURL,
    liveReload: true
}
if (config.dev.browserSync.https) {
    // Add to serverConfig
} else {
    // Add to serverConfig
}
server.init(serverConfig);
`

Comment (miklb) [reply gulpfile.babel.js]: OK, I'll tackle this next. Figured out it was Prettier extension in VS Code that did all of that formatting change.

**Comment by miklb** [comment]
ugh, I did a poor job of backing out my changes and cleaning up the linting. Let me know if you would prefer a fresh PR.

Thanks for the feedback.

**Comment by bamadesigner** [comment]
Hey @miklb! Thanks for the cleanup!

I don't think we need a new PR but would you mind squashing your commits? If you've never squashed before, just let me now and I can walk you through it. It can be confusing.

**Comment by miklb** [comment]
It’s been a while but I’m confident I can figure it out. Thanks again for walking me through contributing

**Comment by miklb** [comment]
did that work?

**Comment by bamadesigner** [comment]
@miklb No. If it was squashed, it would be only 1 commit.

If you're familiar with Terminal, you can do the following:

1. Make sure your branch is up to date with master.
2. Run `git rebase -i master`.
3. You should see a list of commits, starting with `pick`.
4. Make sure the first one says `pick` and change the rest from `pick` to `squash`. Basically, this means you want to squash your commits into the one before it.
5. Save and close the editor.
6. Then it will give you the opportunity to change the commit message. Save and close again.
7. Then you have to force push the new squashed commit: `git push -f`

I hope this was helpful. It's tricky but once you figure it out, it's really helpful.

**Comment by miklb** [comment]
ah, forced push is what I was missing. Thanks! I'll fix.

**Comment by mor10** [comment]
@bamadesigner That rundown of how to squash commits should be added to the Wiki for future reference!

**Comment by mor10** [comment]
@miklb Something about this PR makes the watch tasks in my test environment not work. I am too jetlagged to see what's going on here, but there must be something happening in the gulpfile. To test, simply use this PR, `npm run build`, and make a change to a CSS or PHP file. In my setup, nothing happens, but if I revert to a different commit, it works fine.

@bamadesigner can you confirm?

**Comment by miklb** [comment]
I'll look closer, I was having a difficult time testing locally with the defaults, thus the PR.

**Comment by bamadesigner** [comment]
Thanks for the PR and the hard work @miklb! You were the foundation/inspiration for us to merge #57 and add https support for BrowserSync!

**Pull request closed without merging**